random_image_generator/RandomImageGenerator.py

```diff
@@ -1,193 +1,213 @@
+    
+    def __image_iterator(self, filepath, count):
+        # Format Alignment
+        if filepath[-1] is not "/":
+            filepath += "/"
+        
+        # Allow randomly select n images to save time
+        image_lst = os.listdir(filepath)
+        if count != None and count > 0 and count < len(image_lst):
+            image_lst = sample(image_lst, count)
+
+        # Image iterator
+        for image_id in image_lst:
+            image_path = "".join([filepath, image_id])
+            image = Image.open(image_path)
+            yield image
+
+    def __text_generator(self, dictionary):
+        length = randint(4, 12)
+        text = "".join([dictionary[randint(0, len(dictionary) - 1)] for i in range(length)])
+        return text
+    
+    def __watermark_generator(self):
+        filepath = "watermarks/"
+        image_lst = os.listdir(filepath)
+        # Watermark iterator
+        while True:
+            image_id = sample(image_lst, 1)[0]
+            image_path = "".join([filepath, image_id])
+            image = Image.open(image_path)
+            yield image
+
+    def __add_text_to_image(self, mode, image, text, place):
+        
+        if mode == "ch":
+            fontpath = "./simsun.ttc"  # 宋体
+            font = ImageFont.truetype(fontpath, 32)
+            character_special_resize = 4
+        elif mode == "en":
+            font = None
+            character_special_resize = 1
+        elif mode == "asc":
+            font = None
+            character_special_resize = 1
+        else:
+            print("mode not supported")
+            print("current modes available: en=English, ch=Chinese, asc=ASCII-chars")
+            raise ValueError
+
+        rgba_image = image.convert('RGBA')
+        
+        # Enlarge small images, reduce big images
+        if rgba_image.size[0] < 100:
+            size_factor = 0.3 + random() / 2
+        else:
+            size_factor = 1 + random() * 2
+        size_factor = size_factor / character_special_resize
+        
+        
+        # A separate layer for watermark texts
+        text_overlay = Image.new(
+            'RGBA', 
+            (int(rgba_image.size[0] // size_factor), int(rgba_image.size[1] // size_factor)), 
+            (255, 255, 255, 0)
+        )
+        image_draw = ImageDraw.Draw(text_overlay)
+
+        text_size_x, text_size_y = image_draw.textsize(text, font=font)
+        if place=="random":
+            top_left_corner_x = randint(0, int(rgba_image.size[0] // size_factor - text_size_x))
+            top_left_corner_y = randint(0, int(rgba_image.size[1] // size_factor - text_size_y))
+            
+                   
+        elif place=="corner":
+            top_left_corner_x = rgba_image.size[0] // size_factor - text_size_x
+            top_left_corner_y = rgba_image.size[1] // size_factor - text_size_y
+        text_xy = (top_left_corner_x, top_left_corner_y)
+        
+# output --> txt
+        with open('123.txt', 'wb') as f:

```

Comment:
123 is just an example on Notion. It says "output to txt files OF THE SAME NAMES. (e.g. For a generated image called 123.jpg, we would also want to output a text file called 123.txt containing the bounding box information)"